service/s3.h

```diff
@@ -261,9 +259,21 @@ struct S3Api : public AwsApi {
         std::string contentEncoding;
         std::map<std::string, std::string> metadata;
         std::string acl;
+    };
+
+    /** A set of options that are used when performing uploads. */
+    struct UploadOptions {
```

Comment:
Could we have add the new ACL option to the metadata object?  Why do we divide the options into these two sets?